src/sentry/api/serializers/models/userreport.py

```diff
@@ -12,34 +12,45 @@ def get_attrs(self, item_list, user):
         queryset = list(EventUser.objects.filter(
             id__in=[i.event_user_id for i in item_list],
         ))
+
         event_users = {e.id: d for e, d in zip(queryset, serialize(queryset, user))}
 
+        # If a event list with multiple project IDs is passed to this and event IDs are not unique
+        # this could return the wrong eventIDs
+        events_list = Event.objects.filter(
+            project_id__in={i.project_id for i in item_list},
+            event_id__in=[i.event_id for i in item_list]
+        ).values('id', 'event_id')
+
+        events_dict = {e['event_id']: e['id'] for e in events_list}
+
         attrs = {}
         for item in item_list:
             attrs[item] = {
                 'event_user': event_users.get(item.event_user_id),
+                'event_id': events_dict.get(item.event_id)
             }
+
         return attrs
 
     def serialize(self, obj, attrs, user):
         # TODO(dcramer): add in various context from the event
         # context == user / http / extra interfaces
         return {
-            'id':
-            six.text_type(obj.id),
-            'eventID':
-            obj.event_id,
+            'id': six.text_type(obj.id),
+            'eventID': obj.event_id,
             'name': (
                 obj.name or obj.email or
                 (attrs['event_user'].get_display_name() if attrs['event_user'] else None)
             ),
             'email': (obj.email or (attrs['event_user'].email if attrs['event_user'] else None)),
-            'comments':
-            obj.comments,
-            'dateCreated':
-            obj.date_added,
-            'user':
-            attrs['event_user'],
+            'comments': obj.comments,
+            'dateCreated': obj.date_added,
+            'user': attrs['event_user'],
+            'event': {
+                'id': six.text_type(attrs['event_id']) if attrs['event_id'] else None
```

Comment:
I'm not sure this is the best naming convention. I think it would make sense if event related attributes e.g. `event` and `eventID` were grouped together here, but not sure if there are other consumers of this API we need to think about.

Reply:
🙈